benchmark-framework/src/main/java/io/openmessaging/benchmark/worker/LocalWorker.java

```diff
@@ -217,8 +218,25 @@ public void startLoad(ProducerWorkAssignment producerWorkAssignment) {
 
     @Override
     public void probeProducers() throws IOException {
-        producers.forEach(producer -> producer.sendAsync(Optional.of("key"), new byte[24])
-                .thenRun(() -> totalMessagesSent.increment()));
+        // Asynchronously have our local Producers produce a single message to force connectivity.
+        log.info("beginning probe of {} producers", producers.size());
+        int cnt = producers
+            .parallelStream()
+            .map(p -> p.sendAsync(Optional.of("key"), new byte[24]))
+            .mapToInt(f -> {
+                try {
+                    f.get(30, TimeUnit.SECONDS); // if we take longer than 30s to probe one producer, something is wrong!
+                    totalMessagesSent.increment();
+                } catch (Exception e) {
+                    log.error("error probing producer", e);
+                }
+                return 1;
+            })
+            .sum();
+        // Check our work and report success rate.
+        if (cnt != producers.size()) {
+            log.warn("only probed {}/{} producers", cnt, producers.size());
+        }
```

Comment:
What happens/should we do if we don't probe all producers? This seems like it would fall through and continue working but leave us in an unexpected producer shape?

Reply:
Hmm. Good point. This code path in the `WorkloadGenerator` class relies on exceptions, so I'll change the conditional check at the end. We can't raise an Exception from within the parallel stream processing, so we'll have to do it at the end.

I also think on bumping the timeout makes a sense too. It's still a bit arbitrary and we need to test it out.

Reply:
Might make sense to have this entire section configurable so we can fallback to previous behavior? Also making the timeout configurable.

Reply:
I think you can raise an exception from the stream processing, but it just has to be unchecked (i.e, subclass of `RuntimeException`): a common requirement for a lot of the Java 8 stream stuff.

Reply:
Ah yes, good point. Given we can't guarantee how the stream processing will abort with a runtime exception, it's probably worth just cleanly handling these and raising the `IOException` in my opinion.